File: 3761/Main.py
from typing import List

class Solution:
    def reverseNumber(self, n: int) -> int:
        revNum = 0
        while n > 0:
            revNum = revNum * 10 + (n % 10)
            n //= 10
        return revNum
    
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        indx = {}
        n = len(nums)
        answer = n
        
        for i in range(n):
            if nums[i] in indx:
                answer = min(answer, i - indx[nums[i]] + 1)
            rev = self.reverseNumber(nums[i])
            indx[rev] = i + 1
        
        return answer if answer != n else -1

File: 3761/Main.js
function reverseNumber(n) {
    let revNum = 0;
    while (n > 0) {
        revNum = revNum * 10 + (n % 10);
        n = Math.floor(n / 10);
    }
    return revNum;
}


/**
 * @param {number[]} nums
 * @return {number}
 */
var minMirrorPairDistance = function(nums) {
    const indx = {};
    const n = nums.length;
    let answer = n;

    for (let i = 0; i < n; i++) {
        if (nums[i] in indx) {
            answer = Math.min(answer, i - indx[nums[i]] + 1);
        }
        indx[reverseNumber(nums[i])] = i + 1;
    }

    return answer === n ? -1 : answer;
};


